roles/rsyslog/tasks/main.yaml

```diff
@@ -1,19 +1,19 @@
 ---
+- name: Get required packages
+  set_fact: _rsyslog_packages="{{ (rsyslog_base_packages) + (rsyslog_tls_packages if rsyslog_pki else []) + (rsyslog_additional_packages) }}"
+
 - name: Install/Update required packages
   package:
-    name: "{{ lookup('flattened', rsyslog_base_packages, rsyslog_tls_packages if rsyslog_pki else [], rsyslog_packages) }}"
+    name: "{{ item }}"
     state: latest
+  loop: "{{ _rsyslog_packages | flatten | d('') }}"
```

Comment:
Why this change? https://github.com/oasis-roles/meta_standards#ansible-best-practices says: Avoid calling the package module iteratively with the {{ item }} argument, as this is impressively more slow than calling it with the line name: "{{ foo_packages }}". The same can go for many other modules that can be given an entire list of items all at once.

Reply:
Understood...   I introduced the loop to check whether each package is already installed or not here [1].  (Otherwise, I got an error from package/yum, so I had to put [2] to ignore it.  I thought it'd be better getting rid of [2].)  What'd be the best way to handle that kind of contradictory issue?  Do you recommend just passing the list to the `package` module and setting the ignore error?
[1] https://github.com/linux-system-roles/logging/pull/87/files#diff-ed5a08a2e61430dc18e2d3820cbd6f32R13
[2] https://github.com/linux-system-roles/logging/pull/87/files#diff-ed5a08a2e61430dc18e2d3820cbd6f32L9

Reply:
What error did you get?

Reply:
(and yes, using `ignore_errors` like [2] is usually not a very good idea. But other roles and the best practices have been using package list without a loop and without `ignore_errors`. If there is a problem with applying the best practices, please investigate it and report it, instead of doing things differently.)

Reply:
Hmmm, even though I reverted this change, I cannot reproduce the error...  Maybe, the cause was something else.  Sorry for the noise.  I keep reverting this change for now.

Reply:
Note: I reverted the code pointed out by @pcahyna.